src/vehicle.h

```diff
@@ -716,13 +716,21 @@ class vehicle
 
         units::volume total_folded_volume() const;
 
+        struct fuel_consumption_data {
```

Comment:
Why isn't this just a std::list<int>?
We need a VERY good reason to step away from standard data structures and start building our own, and I don't see a case for that here.

Reply:
No idea. Mark suggested this precise solution.
I'm not c++ programmer so I assumed some esoteric memory/performance issues.

Although later he mentioned in regard to this PR that his knowledge of libraries is out of date.

If there is a reason to change it to a list is there a specific implementation I should use?|
Or just something like 
```cpp
std::list.push( new_val );
max += new_val;
max -= std::list.first();
std::list.pop();
```
?

Reply:
I switched to list